src/D2L.CodeStyle.TestAnalyzers/NUnit/ConfigTestSetupStringsAnalyzer.cs

```diff
@@ -0,0 +1,107 @@
+using System.Collections.Immutable;
+using System.Linq;
+using D2L.CodeStyle.TestAnalyzers.Common;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace D2L.CodeStyle.TestAnalyzers.NUnit {
+
+	[DiagnosticAnalyzer( LanguageNames.CSharp )]
+	public sealed class ConfigTestSetupStringsAnalyzer : DiagnosticAnalyzer {
+
+		private const string AttributeName = "ConfigTestSetupAttribute";
+
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+			=> ImmutableArray.Create( Diagnostics.ConfigTestSetupStrings );
+
+		public override void Initialize( AnalysisContext context ) {
+
+			context.EnableConcurrentExecution();
+			context.RegisterCompilationStartAction( Register );
+		}
+
+		private void Register( CompilationStartAnalysisContext compilation ) {
+
+			compilation.RegisterSyntaxNodeAction(
+					AnalyzeSyntaxNode,
+					SyntaxKind.MethodDeclaration,
+					SyntaxKind.ClassDeclaration
+				);
+		}
+
+		private void AnalyzeSyntaxNode( SyntaxNodeAnalysisContext context ) {
+
+			SyntaxList<AttributeListSyntax> attributeList;
+
+			switch( context.Node ) {
+
+				case MethodDeclarationSyntax method:
+					attributeList = method.AttributeLists;
+					break;
+
+				case ClassDeclarationSyntax @class:
+					attributeList = @class.AttributeLists;
+					break;
+
+				default:
+					return;
+			}
+
+			AttributeSyntax[] attributes = attributeList
+				.SelectMany( x => x.Attributes )
+				.ToArray();
+
+			// Method has no attributes
+			if( !attributes.Any() ) {
+				return;
+			}
+
+			foreach( AttributeSyntax attribute in attributes ) {
+
+				ISymbol symbol = context
+					.SemanticModel
+					.GetSymbolInfo( attribute )
+					.Symbol;
+
+				if( symbol == null ) {
+					continue;
+				}
+
+				// Not a [ConfigTestSetup()]

```

Comment:
The prefered approach is to look up the symbol for ConfigTestSetup in the setup and not register callbacks if it doesn't exist. If it does, compare the strongly typed things rather than strings.

Here's an analyzer that does that kind of stuff: https://github.com/Brightspace/D2L.CodeStyle/blob/master/src/D2L.CodeStyle.Analyzers/ApiUsage/RpcAnalyzer.cs#L25